### ISSUE TITLE
Pixel artifact between sprite, on specific camera position.

### ISSUE THREAD

**Issue by thiolliere** [bevyengine/bevy#4748]
## Bevy version

Bevy 0.7.0 compiled with default features

## Operating system & version

Linux/wayland

## What you did

Draw some sprite from texture atlas:
```rust
use bevy::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .insert_resource(Msaa { samples: 1 })
        .add_startup_system(setup)
        .run();
}
fn setup(
    mut commands: Commands,
    asset_server: Res<AssetServer>,
    mut texture_atlases: ResMut<Assets<TextureAtlas>>,
) {
    let texture_handle = asset_server.load("Minifantasy_TownsAllTiles.png");
    let texture_atlas = TextureAtlas::from_grid(texture_handle, Vec2::new(8.0, 8.0), 64, 36);
    let texture_atlas_handle = texture_atlases.add(texture_atlas);

    commands
        .spawn_bundle(OrthographicCameraBundle::new_2d())
        .insert(Transform {
            translation: Vec3::new(0.0, 230.12491, 0.0),
            scale: Vec3::new(1.0, 0.25, 1.0),
            ..Default::default()
        });

    for y in 0..100 {
        commands.spawn_bundle(SpriteSheetBundle {
            sprite: TextureAtlasSprite::new(66),
            texture_atlas: texture_atlas_handle.clone(),
            transform: Transform {
                translation: Vec3::new(0.0, y as f32 * 8.0, 0.0),
                scale: Vec3::new(100.0, 1.0, 1.0),
                ..Default::default()
            },
            ..Default::default()
        });
    }
}
```

## What you expected to happen

To have all sprite correctly alligned and no pixel in between

## What actually happened

Some artifact are visible:
![Screenshot from 2022-05-14 20-02-46](https://user-images.githubusercontent.com/10416848/168423357-2d69fa5e-fd68-4abc-8b60-9343bc39f602.png)

## Additional information

* This happens at some specific value for the camera position. (In my game when I move randomly it happens every 100 frame or so)

* The issue can be reproduced with this repo: https://github.com/thiolliere/debug_pixel_y_issue

* It is not related to MXAA issues as I set samples to 1

**Comment by thiolliere** [comment]
I tried to narrow the issue:

I now draw from a sprite atlas of 9 pixel: one white surrounded by 8 red.


I draw one rectangle with this sprite, the top pixel of the rectangle is red, instead of white:
![Screenshot from 2022-05-15 12-43-45](https://user-images.githubusercontent.com/10416848/168456308-d457bb05-9c4d-490d-917f-5949a9db800e.png)

(Maybe it is not solvable, I saw some post on the net advising to surround the sprite with some colors to avoid the issue.)

**Comment by thiolliere** [comment]
TLDR: I think it is some issue on the gpu, so nothing to fix in bevy.

looking at the code, there is no float operation resulting in approximation. We directly pass the uv_vertex in the shader.
To fix this kind of issue I easy way is to add some padding in the texture atlas in between tiles. or maybe having the camera always aligned on some pixel, not sure.

**Comment by thiolliere** [comment]
For other people running in the issue, I fixed by putting every tile into its own image, I load the tileset as a single image, and then create one image per tile using the loaded image.

**Comment by lukas-kirschner** [comment]
I managed to fix (or at least work around) the texture bleeding by applying a combination of `ImageSettings`, `FilterMode` and `MSAA`:

```rust
.insert_resource(ImageSettings::default_nearest())
.insert_resource(FilterMode::Nearest)
.insert_resource(Msaa { samples: 1 })
```

Not sure if this workaround works on different systems than my laptop, though.

**Comment by lukas-kirschner** [comment]
After changing the texture resolution to 16x16, the artifacts and the bleeding returned. I managed to fix the problem by rendering the world into a low-resolution texture and displaying this texture as sprite through the main camera. This approach may cause some programming and run-time overhead, but seems to work platform-independently. [This commit](https://github.com/lukas-kirschner/Exodus/commit/534c5ca32935299e134967285dca5f3c7b14c057) shows the changes I had to introduce to my example game.

I adapted some code from [this example](https://github.com/bevyengine/bevy/blob/main/examples/3d/render_to_texture.rs) which might be a good starting point for an implementation.
